testkit/src/main/java/org/apache/calcite/test/RelMetadataFixture.java

```diff
@@ -264,6 +265,24 @@ public RelMetadataFixture assertPercentageOriginalRows(Matcher<Double> matcher)
     return this;
   }
 
+  public RelMetadataFixture assertForeignKeys(
+      Matcher<ImmutableBitSet> constraintMatcher) {
+    RelNode rel = toRel();
+    RelMetadataQuery mq = rel.getCluster().getMetadataQuery();
+    ImmutableBitSet foreignKeys = mq.getForeignKeys(rel, false);
```

Comment:
It seems that we are not covering cases with `ignoreNulls` to `true`, can we add such tests?

Reply:
As the [discussion](https://github.com/apache/calcite/pull/3264#discussion_r1230251674) above, maybe we don't need to add a test where `ignoreNulls` equals `true`.

Reply:
I see your point, but we are still supporting a "mode" we aren't testing, and which comes with bits of code, so it needs proper testing like any other part of our codebase, we should have tests for `ignoreNulls = true` too.

Reply:
OK, got your point, I will add it.